test-cases/python/web-security-issues.py

```
@@ -0,0 +1,152 @@
+def process_image(image_path):
+    command = f"convert {image_path} output.jpg"
+    subprocess.run(command, shell=True)
+
+def hash_password(password):
+    return hashlib.md5(password.encode()).hexdigest()
+
+@app.route('/login', methods=['POST'])
+def login():
+    user_id = request.form.get('user_id')
+    if user_id:
+        session['user_id'] = user_id
```

Ruff (0.12.2)

Undefined name `session`

(F821)